index.js

```diff
@@ -290,7 +290,12 @@ function parse (args, opts) {
   // e.g., --foo apple banana cat becomes ["apple", "banana", "cat"]
   function eatArray (i, key, args) {
     for (var ii = i + 1; ii < args.length; ii++) {
-      if (/^-/.test(args[ii])) break
+      if (/^-/.test(args[ii])) {
+        if (args[ii - 1].replace('-', '') === key) {
```

Comment:
sorry for the slow feedback, finally getting around to merging some things on yargs. I think it might be less fragile to simply use this check:

```js
if (ii === i + 1) {
}
```
ensuring that it's the first iteration of the loop.